### PR TITLE
add OSX arm64 support

### PR DESCRIPTION
**Pull request by matthiasdiener**
Supersedes #32 

closes https://github.com/conda-forge/mesalib-feedstock/pull/39

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Does osx have cdts at all? Maybe you should just do `# [linux]` ???

**Comment by matthiasdiener** [comment]
> Does osx have cdts at all? Maybe you should just do `# [linux]` ???

I think you are right, but what confuses me is that cdts apparently are installable on osx/x86_64...

**Comment by chrisburr** [comment]
@conda-forge-admin, please rerender

**Comment by hmaarrfk** [comment]
You might want to look at https://github.com/conda-forge/harfbuzz-feedstock/blob/master/recipe/build.sh#L25 if you haven't already. It is an other build that uses meson+cross compilation.

File: conda-forge.yml [hunk 1]
```diff
@@ -2,3 +2,5 @@ provider: {linux_aarch64: default, linux_ppc64le: azure}
 conda_forge_output_validation: true
 os_version:
   linux_64: cos7
+build_platform:
+  osx_arm64: osx_64
```
Comment (hmaarrfk) [inline conda-forge.yml:6]:
```suggestion
  osx_arm64: osx_64
test_on_native_only: true
```

**Comment by hmaarrfk** [comment]
lol bad gateway

**Comment by matthiasdiener** [comment]
@conda-forge-admin restart ci

**Comment by matthiasdiener** [comment]
> You might want to look at https://github.com/conda-forge/harfbuzz-feedstock/blob/master/recipe/build.sh#L25 if you haven't already. It is an other build that uses meson+cross compilation.

Thank you, this was the missing piece!

**Comment by hmaarrfk** [comment]
do you want to copy the comment from there as well?

**Comment by hmaarrfk** [comment]
i guess it isn't super appropriate

**Comment by matthiasdiener** [comment]
> i guess it isn't super appropriate

Yeah, I think this isn't directly applicable here.

**Comment by hmaarrfk** [comment]
Thanks!

**Pull request merged**